mountpoint-s3/tests/fuse_tests/write_test.rs

```diff
@@ -435,6 +437,65 @@ where
     assert_eq!(err.raw_os_error(), Some(libc::ENOENT));
 }
 
+fn write_with_storage_class_test<F>(creator_fn: F, storage_class: &str)
+where
+    F: FnOnce(&str, TestSessionConfig) -> (TempDir, BackgroundSession, TestClientBox),
+{
+    const KEY: &str = "new.txt";
+
+    let filesystem_config: S3FilesystemConfig = S3FilesystemConfig {
+        storage_class: Some(storage_class.to_owned()),
+        ..Default::default()
+    };
+
+    let config = TestSessionConfig {
+        filesystem_config,
+        ..Default::default()
+    };
+    let (mount_point, _session, test_client) = creator_fn("write_with_storage_class_test", config);
+
+    let path = mount_point.path().join(KEY);
+
+    let mut f = open_for_write(&path, false).unwrap();
+
+    let data = [0xaa; 16];
+    f.write_all(&data).unwrap();
+    f.sync_all().unwrap();
+    drop(f);
+
+    let s3_storage_class = test_client.get_storage_class(KEY).unwrap();
+
+    assert_eq!(s3_storage_class, storage_class);
+}
+
+fn write_with_invalid_storage_class_test<F>(creator_fn: F, storage_class: &str)
+where
+    F: FnOnce(&str, TestSessionConfig) -> (TempDir, BackgroundSession, TestClientBox),
+{
+    const KEY: &str = "new.txt";
+
+    let filesystem_config: S3FilesystemConfig = S3FilesystemConfig {
+        storage_class: Some(storage_class.to_owned()),
+        ..Default::default()
+    };
+
+    let config = TestSessionConfig {
+        filesystem_config,
+        ..Default::default()
+    };
+    let (mount_point, _session, _test_client) = creator_fn("write_with_storage_class_test", config);
+
+    let path = mount_point.path().join(KEY);
+    write_file(path).expect_err("write with invalid storage class should fail");
+}
+
+fn write_file(path: PathBuf) -> std::io::Result<()> {
```

Comment:
Prefer `path: impl AsRef<Path>` (look at `open_for_write`, for example). 
With `PathBuf` you are requiring callers to give you ownership of the buffer containing the path, while something like `&Path` would just borrow a reference to the path. This is similar to the difference between `String` and `&str`. Here `impl AsRef<Path>` is slightly more complicated, but it means a type that implements the `AsRef<Path>` trait: roughly something that can be converted into a `&Path`.